aten/src/ATen/native/tags.yaml

```diff
@@ -40,3 +40,8 @@
           type promotion and boardcasting ops.
           Canonical aten ops is also effectively the opset produced by torchdynamo.export(aten_graph=True),
           and thus can be used as an opset for export purpose.
+- tag: pointwise
+  desc: |
+          Pointwise operators are operators where each element of the output is computed only by accessing
+          the corresponding element of all the broadcasted inputs. The output shape will be the broadcasted
+          shape of the inputs.
```

Comment:
Does pointwise imply that it is implemented with TensorIterator?

Reply:
Pointwise is more a description of how an operator is computed than its underlying implementation. I would imagine all pointwise operators should be computed with TensorIterator but I don't think that's a necessary condition. Open to input though.

Reply:
The main reason I ask is because there is a bunch of subtle striding behavior, which is unlikely to be done correctly if you're not using TensorIterator under the hood. So, it matters materially for the more subtle invariants whether or not you're buying into "all of the TensorIterator semantics". We don't have to call it TI but I am interested in knowing if we are giving these guarantees.

Reply:
I don't think so. although would be interested to hear from @wanchaol and @Chillee, who have essentially been maintaining their own pointwise op set.

Reply:
To add some context about the DTensor pointwise op set, it was initially added from manual inspection of every op in the native_functions.yaml by referencing https://pytorch.org/docs/stable/torch.html#pointwise-ops, then added some missing pointwise ops when we actually tried on real models. I believe this set is still not completely including all possible pointwise ops yet (but it's close).

For the meaning of pointwise ops, I think the description from @eellison should be a fair enough description of the definition of a pointwise op, but I agree we might need some formal algorithmic guard on it to ensure existing and newly added pointwise op get tagged too.